drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/RightPanelViewImpl.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.scenariosimulation.client.rightpanel;
+
+import javax.enterprise.context.Dependent;
+
+import com.google.gwt.dom.client.DivElement;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.LIElement;
+import com.google.gwt.dom.client.UListElement;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.user.client.ui.Composite;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Dependent
+@Templated
+public class RightPanelViewImpl
+        extends Composite
+        implements RightPanelView {
+
+
+    private Presenter presenter;
+
+    @DataField("rightPanelTabs")
+    private UListElement rightPanelTabs = Document.get().createULElement();
+
+    @DataField("editorTab")
+    protected LIElement editorTab = Document.get().createLIElement();  // protected for test purpose
+
+    @DataField("cheatSheetTab")
+    protected LIElement cheatSheetTab = Document.get().createLIElement();  // protected for test purpose
+
+    @DataField("editorTabContent")
+    protected DivElement editorTabContent = Document.get().createDivElement();  // protected for test purpose
+
+    @DataField("cheatSheetTabContent")
+    protected DivElement cheatSheetTabContent = Document.get().createDivElement(); // protected for test purpose
+
+    public RightPanelViewImpl() {
+
+    }
+
+    @Override
+    public void init(Presenter presenter) {
+        this.presenter = presenter;
+
+    }
+
+    @EventHandler("editorTab")
+    public void onEditorTabClick(ClickEvent event) {
+        showTab(editorTab, editorTabContent);
+        hideTab(cheatSheetTab, cheatSheetTabContent);
+    }
+
+    @EventHandler("cheatSheetTab")
+    public void onCheatSheetTabClick(ClickEvent event) {
```

Comment:
This should call a method in Presenter and that should call showTab/hideTab based on the actions. You can also remove the test that targets this ViewImpl and test the Presenter instead.

Reply:
Hi @Rikkola - 
maybe you're right. But if those fields/methods are private (as they should) how could we test that the different tab are actually shown/hidden depending on the fired event ?  EventHandler is managed by framework (errai) but what we do inside it is done by us....

Reply:
@gitgabrio Here is what I think it should look like. Made a gist out of it. Perhaps easier to read than a comment.

https://gist.github.com/Rikkola/fa097bb4097536ca0cd17c7adaa213d2

Reply:
So each action gets passed back to Presenter so we can test it and ViewImpl is kept as stupid as possible. It might even make sense to add hideCheatSheetTabTab and hideEditorTab methods into the view. So you can test they are hidden.

Reply:
@gitgabrio Ok I updated the gist with the hide methods.

Reply:
Fixed with
[DROOLS-2780] Refactored RightPanelView/RightPanel interaction. Removing access modifier for testing purpose. Renaming mocked instances.